docroot/WEB-INF/src/com/liferay/course/service/impl/CourseServiceImpl.java

```diff
@@ -14,26 +14,86 @@
 
 package com.liferay.course.service.impl;
 
+import java.util.List;
+
+import com.liferay.counter.service.CounterLocalServiceUtil;
+import com.liferay.course.model.Course;
+import com.liferay.course.service.CourseLocalServiceUtil;
 import com.liferay.course.service.base.CourseServiceBaseImpl;
 
 /**
  * The implementation of the course remote service.
  *
  * <p>
- * All custom service methods should be put in this class. Whenever methods are added, rerun ServiceBuilder to copy their definitions into the {@link com.liferay.course.service.CourseService} interface.
+ * All custom service methods should be put in this class. Whenever methods are
+ * added, rerun ServiceBuilder to copy their definitions into the
+ * {@link com.liferay.course.service.CourseService} interface.
  *
  * <p>
- * This is a remote service. Methods of this service are expected to have security checks based on the propagated JAAS credentials because this service can be accessed remotely.
+ * This is a remote service. Methods of this service are expected to have
+ * security checks based on the propagated JAAS credentials because this service
+ * can be accessed remotely.
  * </p>
  *
  * @author ces
  * @see com.liferay.course.service.base.CourseServiceBaseImpl
  * @see com.liferay.course.service.CourseServiceUtil
  */
 public class CourseServiceImpl extends CourseServiceBaseImpl {
+
 	/*
 	 * NOTE FOR DEVELOPERS:
 	 *
-	 * Never reference this interface directly. Always use {@link com.liferay.course.service.CourseServiceUtil} to access the course remote service.
+	 * Never reference this interface directly. Always use {@link
+	 * com.liferay.course.service.CourseServiceUtil} to access the course remote
+	 * service.
 	 */
+
+	public void addCourse(String name, String description, String lecturer, int duration, boolean status)
+			throws Exception {
+
+		Course course = CourseLocalServiceUtil.createCourse(CounterLocalServiceUtil.increment());
```

Comment:
It will be a better choice to use CounterLocalServiceUtil.increment(Course.class.getName()). Try to use counterLocalService too.